### ISSUE TITLE
Custom overlay closes the overlay on mouseClicks which are outside children is Safari and IE 11 

### ISSUE THREAD

**Issue by mikaelrss** [gpbl/react-day-picker#706]
When creating a custom overlay containing a button with a onClick, the overlay component is blurred without triggering the onClick of the button on Safari and IE 11. 

This works fine in 7.0.7, and the bug was introduced in 7.1.0.

Here is a sandbox which illustrates the issue. 
https://codesandbox.io/s/8n57pn6wy2

**Comment by gpbl** [comment]
This thing with overlays and input fields is diabolic 🤨

The issue is caused by these lines:
https://github.com/gpbl/react-day-picker/blob/v7.1.6/src/DayPickerInput.js#L307-L316

```js
 handleInputBlur(e) {
    this.setState({
      showOverlay:
        this.overlayNode && this.overlayNode.contains(e.relatedTarget),
    });
   // snip
```

When the input dispatches the `blur` event, we use the event's `relatedTarget` to understand if the blur has been caused by focusing the overlay or not. If yes, the overlay should remain visible, otherwise it should be hidden. 

We have a merged PR #700 (ex #691) that should address the issue with IE:
https://github.com/gpbl/react-day-picker/pull/700/files#diff-e2a925e665d836a09575f79095191746L307

Basically, it uses a timeout to workaround the problem. I believe this could work for other browser too, but I haven't had yet the time to check it out. Could you @mikaelrss help us? You should pull the master branch and update the `handleInputBlur` to work with other browser as it does for IE (there's a check that should be removed). Some more instructions for contributing are here: http://react-day-picker.js.org/contributing

**Comment by mikaelrss** [comment]
Yeah, I can take a look at it.

**Comment by gpbl** [comment]
This should be fixed in v7.1.7

**Comment by gpbl** [comment]
Please note it is required to pass `{...props}` to the element if you want to keep the interaction with the input field working correctly

```diff
- function CustomOverlay({ classNames, selectedDay, children }) {
+ function CustomOverlay({ classNames, selectedDay, children, ...props }) {
  return (
-    <div className={classNames.overlayWrapper} style={{ marginLeft: -100 }}>
+    <div {...props} className={classNames.overlayWrapper} style={{ marginLeft: -100 }}>
```

I've updated the example: http://react-day-picker.js.org/examples/input-custom-overlay

**Comment by LuisEgan** [comment]
I'm trying to pass a custom prop to the `overlayComponent` but I can't make it work, if I do any of these the `DayPickerInput` disappears on click without updating the selected date or anything:

The custom overlay:
```
function CustomOverlay(props) {
  const { classNames, selectedDay, children, ...otherProps } = props;

  return (
    <div
      className={classNames.overlayWrapper}
      style={{ marginLeft: -100 }}
      {...otherProps}
    >
      <div className={classNames.overlay}>
        <h3>Hello day picker!</h3>
        <p>
          <input />
          <button onClick={() => console.log("clicked!")}>button</button>
        </p>
        <p>
          {selectedDay
            ? `You picked: ${selectedDay.toLocaleDateString()}`
            : "Please pick a day"}
        </p>
        {children}
      </div>
    </div>
  );
}
```

The `DayPickerInput` :
```
<DayPickerInput
  // Doesn't work
  overlayComponent={props => <CustomOverlayO {...props} customProp={customProp}/>}

  // Doesn't work even without custom prop
  overlayComponent={props => <CustomOverlayO {...props} />}
 
  // Doesn't work (for function CustomOverlay(customProp, props) { )
  overlayComponent={CustomOverlay.bind(null, customProp)}
 
  // Works
  overlayComponent={CustomOverlay}
/>
```
